setup/helpers/abs_helper.rb

```diff
@@ -654,24 +660,143 @@ def valid_abs_resource_hosts?(abs_resource_hosts)
       puts
     else
 
-      begin
-        hosts = JSON.parse(abs_resource_hosts)
-        host = hosts[0]
-        hostname = host["hostname"]
-        if !hostname.nil? && !hostname.empty?
-          is_valid = true
-        else
-          puts "The specified resource host array is not valid: #{abs_resource_hosts}"
-          puts
-        end
-      rescue StandardError
-        # TODO: raise?
-        puts "JSON::ParserError encountered parsing the hosts array: #{abs_resource_hosts}"
+      # begin
+      #   hosts = JSON.parse(abs_resource_hosts)
+      #   host = hosts[0]
+      #   hostname = host["hostname"]
+      #   if !hostname.nil? && !hostname.empty?
+      #     is_valid = true
+      #   else
+      #     puts "The specified resource host array is not valid: #{abs_resource_hosts}"
+      #     puts
+      #   end
+      # rescue StandardError
+      #   # TODO: raise?
+      #   puts "JSON::ParserError encountered parsing the hosts array: #{abs_resource_hosts}"
+      # end
+
+      hosts = parse_abs_resource_hosts(abs_resource_hosts)
```

Comment:
Why is this taken out of the rescue block?  There seem like a number of things that should be caught here (e.g. `parse_abs_resource_hosts` returns nil).

Reply:
I've brought it back.